File: Autobase/Backend/Autobase/src/main/java/com/example/Autobase/dto/request/car/CarUpdateDto.java
package com.example.Autobase.dto.request.car;


import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.Data;

@Data
public class CarUpdateDto {
    private String model;
    private String mark;
    private String color;
    @Positive(message = "Год выпуска должен быть положительным числом")
    private Integer yearOfRelease;
    private String stateNumber;
    @PositiveOrZero(message = "Пробег не может быть отрицательным")
    private Integer mileage;
    private String notes;
    private String status;
}

File: Autobase/Frontend/Autobase/src/app/models/dto/request/car/CarUpdateDto.ts
import { CarStatus } from "../../../enums/CarStatus";

export interface CarUpdateDto {
  model?: string;
  mark?: string;
  color?: string;
  yearOfRelease?: number;
  stateNumber?: string;
  mileage?: number;
  notes?: string;
  status?: CarStatus;     
}